workitem/workitem_repository.go

```diff
@@ -88,10 +88,10 @@ type WorkItemRepository interface {
 	LoadBatchByID(ctx context.Context, ids []uuid.UUID) ([]*WorkItem, error)
 	LoadByIteration(ctx context.Context, id uuid.UUID) ([]*WorkItem, error)
 	LookupIDByNamedSpaceAndNumber(ctx context.Context, ownerName, spaceName string, wiNumber int) (*uuid.UUID, *uuid.UUID, error)
-	Save(ctx context.Context, spaceID uuid.UUID, wi WorkItem, modifierID uuid.UUID) (*WorkItem, error)
+	Save(ctx context.Context, spaceID uuid.UUID, wi WorkItem, modifierID, revisionID uuid.UUID) (*WorkItem, error)
 	Reorder(ctx context.Context, spaceID uuid.UUID, direction DirectionType, targetID *uuid.UUID, wi WorkItem, modifierID uuid.UUID) (*WorkItem, error)
-	Delete(ctx context.Context, id uuid.UUID, suppressorID uuid.UUID) error
-	Create(ctx context.Context, spaceID uuid.UUID, typeID uuid.UUID, fields map[string]interface{}, creatorID uuid.UUID) (*WorkItem, error)
+	Delete(ctx context.Context, id, suppressorID, revisionID uuid.UUID) error
```

Comment:
Is this necessary? As of now, I can see notifications are enabled only for:
1. workitem creation - (https://github.com/fabric8-services/fabric8-wit/blob/master/notification/notification.go#L39)
2. workitem deletion - (https://github.com/fabric8-services/fabric8-wit/blob/master/notification/notification.go#L44)

May be, eventually we would have notification for deletion.. but as of now, do we REALLY need the revision ID for delete workitem as a part of this PR?

Reply:
Right now the code where the revision is actually created needs revisionID and it is being called from all methods `Create`, `Delete` & `Save`.